lib/rubocop/cop/style/it_block_parameter.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Check for replacement with or disallowing of the `it` block parameter.
+      #
+      # It provides three `EnforcedStyle` options:
+      #
+      # 1. `allow_named_parameter` (default) ... Detects only numbered block parameters.
+      # 2. `always` ... Always uses the `it` block parameter.
+      # 3. `disallow` ... Disallows the `it` block parameter.
+      #
+      # A single numbered parameter is detected when `allow_named_parameter` or `always`.
+      #
+      # @example EnforcedStyle: allow_named_parameter (default)
+      #   # bad
+      #   block { do_something(_1) }
+      #
+      #   # good
+      #   block { |it| do_something(it) }
+      #   block { |named_param| do_something(named_param) }
+      #
+      # @example EnforcedStyle: always
+      #   # bad
+      #   block { do_something(_1) }
+      #   block { |named_param| do_something(named_param) }
+      #
+      #   # good
+      #   block { |it| do_something(it) }
+      #
+      # @example EnforcedStyle: disallow
+      #   # bad
+      #   block { |it| do_something(it) }
+      #
+      #   # good
+      #   block { do_something(_1) }
+      #   block { |named_param| do_something(named_param) }
+      #
+      class ItBlockParameter < Base
+        include ConfigurableEnforcedStyle
+        extend TargetRubyVersion
+        extend AutoCorrector
+
+        MSG_USE_IT_BLOCK_PARAMETER = 'Use `it` block parameter.'
+        MSG_AVOID_IT_BLOCK_PARAMETER = 'Avoid using `it` block parameter.'
+
+        minimum_target_ruby_version 3.4
+
+        def on_block(node)
+          return unless style == :always
+          return unless node.arguments.one?
```

Comment:
could maybe bail early if the single arg is a (kw)splat so we don't waste time scanning the block body
```ruby
block { |*args| ... }
block { |**kwargs| ... }
```